src/reflection.jl

```diff
@@ -118,9 +124,21 @@ function code_llvm(io::IO, @nospecialize(job::CompilerJob); optimize::Bool=true,
     # NOTE: jl_dump_function_ir supports stripping metadata, so don't do it in the driver
     str = JuliaContext() do ctx
         ir, meta = codegen(:llvm, job; optimize=optimize, strip=false, validate=false, ctx, kwargs...)
-        ccall(:jl_dump_function_ir, Ref{String},
-              (LLVM.API.LLVMValueRef, Bool, Bool, Ptr{UInt8}),
-              meta.entry, !raw, dump_module, debuginfo)
+        if VERSION >= v"1.9.0-DEV.516"
+            ts_mod = ThreadSafeModule(ir; ctx)
+            # N.B. jl_dump_function_ir will `Libc.free` the passed-in pointer
```

Comment:
@pchintalapudi this is an odd design choice. Normally on C-API we do caller managed memory, so that we could pass in a stack allocation

Reply:
Any update here? This is indeed not a good API; why do we want RAII destruction of all this memory (the argument struct and the module too?) in `jl_dump_function_ir`?